xunit.runner.worker/Connection.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace xunit.runner.worker
+{
+    internal abstract class Connection : IDisposable
+    {
+        private bool _closed;
+
+        internal abstract Stream Stream { get; }
+
+        internal abstract void WaitForClientConnect();
+        internal abstract void WaitForClientDone();
+
+        protected virtual void DisposeCore()
+        {
+
+        }
+
+        private void Dispose()
```

Comment:
I can see that if you want different accessibility, but in this case it looks like just another method.  Any reason `Dispose` should *not* be public here?  I generally frown on explicit implementations unless there is a reason to hide the member.